include/zephyr/toolchain/common.h

```diff
@@ -106,6 +107,38 @@
 
 #endif /* _ASMLANGUAGE */
 
+/*
+ * General directive for assembly code, to align the following symbol, in bytes.
+ *
+ * Example:
+ *
+ *    BALIGN(4)
+ *    test_symbol:
+ *
+ * 'test_symbol' will get aligned to 4 bytes.
+ */
+
+#ifdef _ASMLANGUAGE
```

Comment:
If you use:
```suggestion
#if defined(_ASMLANGUAGE) && !defined(_LINKER)
```
then it will be defined for the linker script template pre-processin, and thus not conflict with the linker directive of same name and you can use `ALIGN` as macro name.

Reply:
Thanks for the pointer. I have updated it.